File: framework/ios/Example/Example/OpenpathWrapper/Models/SyncUserResponse.swift
import Foundation

struct SyncUserResponse: Codable, CustomStringConvertible {
    let user: User
    typealias User = OpenpathUser
    let appVersion: AppVersion?
    struct AppVersion: Codable {
        let id: Int
        let latestBuild: Int
        let latestVersion: String
        let deprecatedBuild: Int
        let deprecatedVersion: String
        let latestSdkVersion: String
        let deprecatedSdkVersion: String
        let os: String
        let isUpdateRequired: Bool
    }

    var description: String {
        let jsonEncoder = JSONEncoder()
        jsonEncoder.outputFormatting = .prettyPrinted
        return try! jsonEncoder.encode(self).description
    }
}

File: types/Events/SyncUserResponse.ts
import { User } from '../Models/User';
import { Credential } from '../Models/Credential';

/**
 * The data included with the "onSyncUserResponse" event.
 *
 * Invoked when {@link syncUser} is complete.
 *
 * @event
 * @see {@link OpenpathEvent}
 * @see syncUser
 */
export interface SyncUserResponse {
  credential: Credential;
  user: User;
  appVersion?: {
    deprecatedBuild: number;
    deprecatedSdkVersion: string;
    deprecatedVersion: string;
    id: number;
    isUpdateRequired: Boolean;
    latestBuild: number;
    latestSdkVersion: string;
    latestVersion: string;
    os: string;
  };
}
